lib/components/Box.tsx

```diff
@@ -63,21 +70,13 @@ type InternalProps = {
    * 3. This should be a static string with minimal interpolation. If you need more logic, prefer the props approach.
    */
   tw: string;
-};
+}>;
 
-// You may wonder why we don't just use ComponentProps<typeof Box> here.
-// This is because I'm trying to isolate DangerDoNotUse from the rest of the props.
-// While you still can technically use ComponentProps, it won't throw an error if someone uses dangerouslySet.
-export type BoxProps<TElement = HTMLDivElement> = Partial<
-  InternalProps & BooleanStyleMap & StringStyleMap & EventHandlers<TElement>
->;
-
-// Don't you dare put this elsewhere
-type DangerDoNotUse = {
-  dangerouslySetInnerHTML?: {
-    __html: any;
-  };
-};
+export interface BoxProps<TElement = HTMLDivElement>
+  extends Omit<InternalProps, 'dangerouslySetInnerHTML'>,
+    BooleanStyleMap,
+    StringStyleMap,
+    EventHandlers<TElement> {}
```

Comment:
this is wacky but it looks better in the editor now
![image](https://github.com/user-attachments/assets/08dedfe4-2717-46a2-87f9-def9d469e0c7)